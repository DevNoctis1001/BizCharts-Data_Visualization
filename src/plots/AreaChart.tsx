import 'react';
import { Area, AreaOptions } from '@antv/g2plot/lib/plots/area';

import createPlot from '../createPlot';

export { AreaOptions };
export default createPlot<AreaOptions>(Area, 'AreaChart');
