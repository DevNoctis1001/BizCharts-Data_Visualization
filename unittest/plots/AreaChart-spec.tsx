import React, { useEffect, useState } from 'react';
import AreaChart from '../../src/plots/AreaChart';
import { render, act } from '@testing-library/react';
import { toMatchDiffSnapshot } from 'snapshot-diff';

// @ts-ignore
global.expect.extend({ toMatchDiffSnapshot });

const MOCK_DATA = [{
  "Date": "2010-01",
  "scales": 1998
},
{
  "Date": "2010-02",
  "scales": 1850
},
{
  "Date": "2010-03",
  "scales": 1720
}]

const demoContainer = document.createElement('div')
const demoContainer2 = document.createElement('div')

const Chart = () => {
  const [data, setData] = useState([]);
  const option = {
    xField: 'Date',
    yField: 'scales',
    xAxis: {
      range: [0, 1],
      tickCount: 5,
    },
  }
  useEffect(() => {
    act(() => setData(MOCK_DATA));
  }, [])
  return <AreaChart data={data} {...option} />
}
demoContainer2.className = 'demoContainer';
document.body.appendChild(demoContainer);

describe('AreaChart', () => {
  test('renders App component', () => {
    const { asFragment, debug } = render(<Chart />, {
      container: demoContainer,
    });
    debug();
    const firstRender = asFragment();
    expect(firstRender).toMatchDiffSnapshot(asFragment());
  });
});
