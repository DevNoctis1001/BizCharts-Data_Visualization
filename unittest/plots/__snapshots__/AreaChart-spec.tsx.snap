// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AreaChart renders App component 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
